pigeons/alarm_api.dart

```diff
@@ -0,0 +1,212 @@
+  ///
+  /// Some devices will need the Autostart permission to show the full screen
+  /// notification. You can check if the permission is granted and request it
+  /// with the [auto_start_flutter](https://pub.dev/packages/auto_start_flutter)
+  /// package.
+  final bool androidFullScreenIntent;
+}
+
+/// Model for notification settings.
+class NotificationSettingsWire {
+  /// Constructs an instance of [NotificationSettingsWire].
+  ///
+  /// Open PR if you want more features.
+  const NotificationSettingsWire({
+    required this.title,
+    required this.body,
+    this.stopButton,
+    this.icon,
+  });
+
+  /// Title of the notification to be shown when alarm is triggered.
+  final String title;
+
+  /// Body of the notification to be shown when alarm is triggered.
+  final String body;
+
+  /// The text to display on the stop button of the notification.
+  ///
+  /// Won't work on iOS if app was killed.
+  /// If null, button will not be shown. Null by default.
+  final String? stopButton;
+
+  /// The icon to display on the notification.
+  ///
+  /// **Only customizable for Android. On iOS, it will use app default icon.**
+  ///
+  /// This refers to the small icon that is displayed in the
+  /// status bar and next to the notification content in both collapsed
+  /// and expanded views.
+  ///
+  /// Note that the icon must be monochrome and on a transparent background and
+  /// preferably 24x24 dp in size.
+  ///
+  /// **Only PNG and XML formats are supported at the moment.
+  /// Please open an issue to request support for more formats.**
+  ///
+  /// You must add your icon to your Android project's `res/drawable` directory.
+  /// Example: `android/app/src/main/res/drawable/notification_icon.png`
+  ///
+  /// And pass: `icon: notification_icon` without the file extension.
+  ///
+  /// If `null`, the default app icon will be used.
+  /// Defaults to `null`.
+  final String? icon;
+}
+
+/// Errors that can occur when interacting with the Alarm API.
+enum AlarmErrorCode {
+  unknown,
+
+  /// A plugin internal error. Please report these as bugs on GitHub.
+  pluginInternal,
+
+  /// The arguments passed to the method are invalid.
+  invalidArguments,
+
+  /// An error occurred while communicating with the native platform.
+  channelError,
+
+  /// The required notification permission was not granted.
+  ///
+  /// Please use an external permission manager such as "permission_handler" to
+  /// request the permission from the user.
+  missingNotificationPermission,
+}
+
+@HostApi()
+abstract class AlarmApi {
+  void setAlarm({required AlarmSettingsWire alarmSettings});
+
+  void stopAlarm({required int alarmId});
+
+  bool isRinging({required int? alarmId});
+
+  void setWarningNotificationOnKill({
+    required String title,
+    required String body,
+  });
+
+  void disableWarningNotificationOnKill();
+}
+
+// TODO(orkun1675): Consider migrating to Pigeon type-safe channels when they
```

Comment:
TODO

Reply:
Removed, there is no reason to use streams to support this API signature.